algobot/helpers.py

```diff
@@ -40,6 +45,89 @@
 LONG_INTERVAL_MAP = {v: k for k, v in SHORT_INTERVAL_MAP.items()}
 
 
+class AppDirTemp(AppDirs):
+    def __init__(self):
+        self.root = tempfile.mkdtemp()
+
+    @property
+    def user_data_dir(self):
+        return os.path.join(self.root, "UserData")
+
+    @property
+    def user_log_dir(self):
+        return os.path.join(self.root, "UserLog")
+
+    @property
+    def site_data_dir(self):
+        return os.path.join(self.root, "SiteData")
+
+    @property
+    def user_config_dir(self):
+        return os.path.join(self.root, "UserConfig")
+
+    @property
+    def site_config_dir(self):
+        return os.path.join(self.root, "SiteConfig")
+
+    @property
+    def user_cache_dir(self):
+        return os.path.join(self.root, "UserCache")
+
+    @property
+    def user_state_dir(self):
+        return os.path.join(self.root, "UserState")
+
+
+class Paths:
+    """ Encapsulates all the path information for the app to store its configuration. """
+    def __init__(self, root_dir: str, app_dirs: AppDirs):
+        self.root_dir = root_dir
+        self.app_dirs = app_dirs
+
+    def get_ui_dir(self) -> str:
+        return os.path.join(self.root_dir, 'UI')
+
+    def get_log_dir(self) -> str:
+        return os.path.join(self.app_dirs.user_log_dir, 'Logs')
+
+    def get_database_dir(self) -> str:
+        return os.path.join(self.app_dirs.user_data_dir, 'Databases')
+
+    def get_state_path(self) -> str:
+        return os.path.join(self.app_dirs.user_data_dir, 'state.json')
+
+    def get_optimizer_results_dir(self) -> str:
+        return os.path.join(self.app_dirs.user_data_dir, 'Optimizer Results')
+
+    def get_backtest_results_dir(self) -> str:
+        return os.path.join(self.app_dirs.user_data_dir, 'Backtest Results')
+
+    def get_trade_history_dir(self) -> str:
+        return os.path.join(self.app_dirs.user_data_dir, 'Trade History')
+
+    def get_volatility_results_dir(self) -> str:
+        return os.path.join(self.app_dirs.user_data_dir, 'Volatility Results')
+
+    def get_csv_dir(self) -> str:
+        return os.path.join(self.app_dirs.user_data_dir, 'CSV')
+
+    def get_configuration_dir(self) -> str:
+        return os.path.join(self.app_dirs.user_data_dir, 'configuration')
```

Comment:
I believe configuration should be capitalized

Reply:
Good catch